### ISSUE TITLE
No sync if file set to unreachable

### ISSUE THREAD

**Issue by CrysK** [keepassium/KeePassium#352]
If you set a kdbx file as "Consider file as inaccessible: immediately", it will never be synchronized. 
I would have expected the file to be synchronized when manually pushing (pull down the screen until the loading circle appears).

Is there a way to synchronize the kdbx file manually only?
I don't want to synchronize the file all the time, but also not never.

iPhone 11 Pro
iOS latest version

**Comment by keepassium** [comment]
> I would have expected the file to be synchronized when manually pushing (pull down the screen until the loading circle appears).

It does exactly this, unless timeout is set to "Immediately". That option is specifically intended to suppress automatic sync.

> Is there a way to synchronize the kdbx file manually only?

Not currently. The corresponding feature request is on the roadmap, but quite deep in the nice-to-have list.

> I don't want to synchronize the file all the time, but also not never.

Can you elaborate on why you don't want automatic sync? Is it data consumption concerns, sync delays, something else?

**Comment by CrysK** [comment]
> Can you elaborate on why you don't want automatic sync? Is it data consumption concerns, sync delays, something else?

- In the vast majority of sync cases (certainly 99%) there is no change to the file. The sync is therefore unnecessary. 
- The average download rate here in Germany is 1 MBit (public networks, train, mobile network, ...). Here a 500kB file takes about 4s to sync. With latency rather up to 9s. And then this is often not enough and it is canceled, although you have set the time to 10s. And always waiting 10s is already very long.
- The smartphone often thinks that you are connected to the Internet, but you are not (completely). For example, because you are in a company network where many private sites are blocked (usually in my case). Or if you have not yet agreed to network regulations for public networks. Here you always wait 10 seconds, even though you know that these 10 seconds are simply unnecessary.

**Comment by keepassium** [comment]
Thank you for the details, @CrysK! (And sorry for the slow response…)

I'm afraid the app architecture does not foresee sync on manual demand, it is either automatic or completely manual. Changing this won't be fast, given the limited demand.

So possible options for the moment are:
- Reduce database timeout from 10s. This way, the app will attempt syncing, and will either succeed (on a fast network) or fall back to the cached copy only after a few seconds.
- Use proper manual sync, that is, manually export the file from your storage provider app to KeePassium. After changes, manually export it back. Between exports, KeePassium will treat this local file as your main database (not a read-only cache). If the file changes rarely, exporting would take less time than waiting for timeouts.

**Comment by CrysK** [comment]
... or simply click on the kdbx file in the app, "Database settings", and under "Consider file as inaccessible" switch from "Immediately" to "in x seconds", synchronize once and then switch back again.
This is much faster than the other two methods described. But it is manual.

Pity that it is not planned or scheduled, but thanks for the answer anyway.
